src/odemis/driver/technolution.py

```diff
@@ -719,45 +710,72 @@ def getTicksDwellTime(self):
-        ASM API the scan offset is defined as the median value of the setpoints during the scanning phase.
-
-        :return (tuple of ints): slope found during the scanning phase for x, y in volts (with range VOLT_RANGE).
-        """
-        scan_offset_volts = convertRange(self.scanOffset.value, numpy.array(self.scanOffset.range)[:, 1], VOLT_RANGE)
-        scan_gain_volts = convertRange(self.scanGain.value, numpy.array(self.scanGain.range)[:, 1], VOLT_RANGE)
-        return tuple(((scan_gain_volts + scan_offset_volts) / 2))
-
-    def getScanGainVolts(self):
+    def getCenterScanVolt(self):
         """
-        Get the scan gain value in Volts as defined in the ASM API.
-        In this wrapper the scan gain as the maximum scan value on a range from -1 to 1, while in the
-        ASM API the scan gain is defined as the gradient of the setpoints during the scanning phase.
+        Calculate the center of the scanning ramp in volt. Convert from arbitrary units to volts.
+        The scan offset is defined as the start of the scanning ramp (typically a sawtooth signal),
+        while the acquisition server (ASM) handles the offset as the center of the scanning ramp.
 
-        :return (tuple of ints): slope found during the scanning phase for x, y in volts (with range VOLT_RANGE).
+        :return (tuple of floats): Center (offset) of the scanning ramp in x and y direction in volt.
         """
         ##################################################################
-        # For the acquisition in the ASM API the scan gain is defined as the stepsize in volts per setpoints.
-        # From the first to the last pixels the number of steps are: number of points - 1.
-        # So: number of steps = resolution - 1
-        # This means that the stepsize in volts per setpoint is (maximum value - minimum value) / number of steps
-        # example: scanning ramp resembled by 4 setpoints (values: 2 to 8)
-        #               --8-- amplitude (scan_gain) -> 4th (last) setpoint
+        # example: scanning ramp resembled by 4 pixels (values: 2 to 8)
+        #               --8-- end -> 4th (last) pixel
         #          --6--
         #     --4--
-        # --2-- offset (scan_offset) -> 1st setpoint
-        # Number of setpoints = 4
-        # Number of steps = number of setpoints - 1 = 3
-        # step size = (8 - 2) / 3 = 6 / 3 = 2
-        #################################################################
-        scan_offset_volts = convertRange(self.scanOffset.value, numpy.array(self.scanOffset.range)[:, 1], VOLT_RANGE)
-        scan_gain_volts = convertRange(self.scanGain.value, numpy.array(self.scanGain.range)[:, 1], VOLT_RANGE)
+        # --2-- start = offset (scan_offset)
+        # end = start + amplitude = 2 + 6 = 8
+        # center =  start + end = 2 + 8 = 5
+        ##################################################################
+        # Convert start of the scanning ramp from [a.u.] to [bits].
```

Comment:
The units are sort of clear, but just be aware that "bits" is not the normal usage. "bit" is normally a unit of storage capacity. Each value here is stored *on* 16 bits. I think that, to be very correct, the original unit is "arbitrary unit mapped between -1 and 1", and the end unit is "arbitrary unit mapped between -2¹⁵ and 2¹⁵".

Reply:
True. I will see if I can clarify this somewhere.